### ISSUE TITLE
Recipe Tags

### ISSUE THREAD

**Issue by gthole** [gthole/drink-stash#10]
- Tags should be manually entered
- Anyone can add / remove tags
- The list of available tags should be predefined by admins
- Recipe List view should support filtering by tags, a different mechanism from searching
- Tags should be displayed on the Recipe List items, alongside or after the ingredient list.
- Tags can be added to a Recipe in the Create/Edit View or on the Recipe Details view.

**Comment by gthole** [comment]
Heya @jelloeggy - I added support for tags!  Want to add some tags and review it?

Notes:
- You can add available tag values in the admin portal.  I added a few just for testing.
- For now, you can filter on tags by doing `tags = <whatever>` in the search bar.
- When you add tags in the admin panel you should reload your browser, since the app caches them (since they're unlikely to change much once we have a good list).
- For various reasons, I swapped the "Edit" and "Favorite" buttons on the recipe page.
- There are a number of styling things that aren't exactly right yet, but let me know if you see anything seriously busted.

**Comment by jelloeggy** [comment]
Much excitement! 
I absolutely look forward to playing with this feature, however the first thing I notice is that I think this recent deploy broke the activity feed. I added a recipe last night and don't see it listed in the feed, even log out/log in and closing my browser.
I noticed the feed because we have a new interested party who I know used it last night and provided a bunch of feedback (!!!!) so I've been wanting to stalk, er, monitor his usage.

**Comment by gthole** [comment]
Oh dang, you're right!  I combined the activity feed and comments views into the same component, and I had to hard-refresh to get the new recipe and comment in my feed.  I'll take a look at fixing it tonight.

I'm excited for a new user, and _very_ curious about the feedback you got.

**Comment by gthole** [comment]
I think I have the activity feed thing all sorted out (and actually somewhat improved) if you want to check and confirm.

**Comment by jelloeggy** [comment]
Do you expect drinks added in the meantime to show up or only going forward? 

I will add in additional comments and feedback soon!

**Comment by jelloeggy** [comment]
Also gave some others a view tonight and may have more interested users! ;-|

**Comment by gthole** [comment]
More interested users, huh?  Sounds like I should finally install Splunk on my server setup.

Reloading your browser will get the new UI code, which should show all the activity going back.  I see the Winter's Stake recipe you added, for example (which looks amazing.)  Our new user hasn't added or done anything in the app yet, though.

**Comment by jelloeggy** [comment]
I see it now! Feed issue confirmed fixed.

**Comment by jelloeggy** [comment]
OK, I tested out some tags. First bug is that the tags appear in the left side recipe list view when they are first added, but they don't show in that view on any subsequent refresh. See attached - all recipes in the view have tags added, but only the most recently added are showing in the list view.
![Screenshot 2019-03-15 at 7 45 21 PM](https://user-images.githubusercontent.com/45017933/54467384-23385080-475b-11e9-8fba-d49be133ca5b.png)

I really like being able to see the tags in that view. Also, I tested searching (i.e. tag = no ice) and that worked nicely.

**Comment by gthole** [comment]
That bug should be squashed.  It was because I had seriously screwed up the caching strategy.  (Which might also have explained the activity feed stuff that I spent so long fixing.)

**Comment by jelloeggy** [comment]
It may be that you are mid-deploy so things are unstable, but it seems like this is not squashed, I still don't see tags in the left panel view. Dig the new picture though! (But Liquor cabinet is misspelled?)

**Comment by gthole** [comment]
You may need to clear your cache?  I definitely see the tags in the left panel after a reload (though I'm clearly starting to run out of ideas why you don't.)

(D'oh!  Thanks for the catch.  Feel free to wordsmith that text too if you think it can be better.)

**Comment by jelloeggy** [comment]
You got it! Took a full cache clearing but its good now. 

It is a very rare day indeed I can correct a spelling error for you so I'll take it. I wish there were better tags I could think of than "no ice" and "on ice" so if you think of any to represent those states let me know before we go too far with tagging recipes.

**Comment by gthole** [comment]
Didn't you have a list of tags that you wrote?  I like "served up" for chilled with ice but not "on ice".  (You should be able to just change the name of the "no ice" tag you made, but again with the browser reloads because tags get cached in the app.)

But I have to say this "served up" / "no ice" tagging seems like a lot of manual work.  Granted you're way ahead of me on data entry at this point, but still.

**Comment by jelloeggy** [comment]
Yes, I referenced the original drink stash notes..."on ice" and "no ice"
were the best I had to try to be accurate without the confusion of "served
up" which apparently also indicates drinkware (to some). I'd be happy to
just mutually agree "served up" means "made with ice and strained".

Tags are best efforts. I don't see it as a holistic project, just value
add. But yes, it would take a lot of work to get 100% coverage.
